cardano-cli/test/cardano-cli-test-lib/Test/Cardano/CLI/Aeson.hs

```diff
@@ -118,3 +121,43 @@ assertHasMapping file obj key value = GHC.withFrozenCallStack $ do
        H.failure
 
 
+-- | @assertEqualModuloDesc file1 file2@ loads @file1@ and @file2@ from disk,
+-- then it strips the field @description@ from the loaded content, and finally compare
+-- the two values. The values must be equal, otherwise the test is failed.
+--
+-- Required, because command @"key" "verification-key"@ generates keys without descriptions.
+-- Note that it would be better to write descriptions, see:
+-- https://github.com/IntersectMBO/cardano-cli/issues/429#issuecomment-2003880575
+assertEqualModuloDesc :: ()
+  => (HasCallStack, MonadIO m, MonadTest m)
+  => FilePath -- ^ The file of the first generated verification key
+  -> FilePath -- ^ The file of the second generated verification key, i.e. the one
+              --   generated by calling "key verification-key"
+  -> m ()
+assertEqualModuloDesc file1 file2 =  GHC.withFrozenCallStack $ do
+    value1 <- H.readJsonFileOk @Value file1
+    value1' <- removeDescription value1
+
+    value2 <- H.readJsonFileOk @Value file2
+    value2' <- removeDescription value2
+
+    value1' H.=== value2'
+
+-- | Removes the @description@ field from a JSON object.
+removeDescription :: ()
+  => (HasCallStack, MonadTest m)
+  => Value
+  -> m Value
+removeDescription v =
+  case v of
+    Object inner ->
+      return $ Object $ Aeson.KeyMap.delete (Aeson.fromText "description") inner
+    Array _  -> failWrongType "array"
+    Number _ -> failWrongType "number"
+    Bool _   -> failWrongType "bool"
+    String _ -> failWrongType "string"
+    Null     -> failWrongType "null"
+   where
+     failWrongType got = do
+       H.note_ $ "Expected object but got: " <> got
+       H.failure
```

Comment:
Do I assume right that the lens will not fail if the `Value` is not an `Object`? (it will simply do nothing)

That's just for me to know, behavior is fine for me in this case.

Reply:
I think so. That's how I understand the signature. Something similar to [`alter`](https://hackage.haskell.org/package/containers-0.7/docs/Data-Map-Internal.html#v:alter).

Reply:
@carbolymer> this would be the first time we use `lens` and `lens-aeson` in this repo, which I'm not comfortable to add myself. It's quite something we are pulling here, so I rather keep my lowtech initial version.